### PR TITLE
Add Source YAML Editor

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR introduces a YAML editor to the Source workspace. This includes 3 major items:
1. Introduce the YAML editor ([Notion PRD](https://www.notion.so/rilldata/Introduce-YAML-editor-to-the-source-experience-99ebcb56cfd3463fabf406ad4d8b2e3a))
2. Add/revisit relevant CTAs ([Notion PRD](https://www.notion.so/rilldata/Source-CTA-buttons-efd7d27ba44a4340964a1cb8820746e9))
3. Properly display error messages ([Notion PRD](https://www.notion.so/rilldata/Error-UI-for-sources-output-pane-f5b14501b9914a71a7f05fdbafbce311))

This PR does _not_ modify the navigation from the Add Source modal ([Notion PRD](https://www.notion.so/rilldata/Adapt-add-source-modal-to-generate-source-config-when-errors-are-present-8f8e5738b65b4cc39ea475345b86f498)). That will be handled in a follow-on PR.

### CODE REVIEW

File: web-local/test/ui/sources.spec.ts [hunk 1]
```diff
@@ -52,4 +61,40 @@ describe("sources", () => {
...
+    await playwrightExpect(page.getByText("file does not exist")).toBeVisible();
+
+    // Edit source path to an existent file
+    const adImpressionsSource = `type: local_file
+path: ${TestDataPath}/AdImpressions.tsv`;
+    await updateCodeEditor(page, adImpressionsSource);
+    const tableRowsPromise = page.waitForResponse(
+      // (We're editing the AdBids source, though the new data is from AdImpressions)
+      new RegExp(`/queries/rows/tables/AdBids`)
+    );
```
Comment (ericpgreen2) [inline web-local/test/ui/sources.spec.ts:88]: Avoid `waitForResponse`

File: web-common/src/features/sources/selectors.ts [hunk 2]
```diff
@@ -60,3 +62,23 @@ export function useEmbeddedSources(instanceId: string) {
     }
   );
 }
+
+export function useIsSourceUnsaved(
+  instanceId: string,
+  sourceName: string,
+  // Include clientYAML in the function call to force the selector to recompute when it changes
+  clientYAML: string
+) {
```
Comment (ericpgreen2) [inline web-common/src/features/sources/selectors.ts:71]: Another option is to pass in the `sourceStore` and compute a `derived` result.

File: web-common/src/features/sources/saveAndRefresh.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { getFilePathFromNameAndType } from "../entity-management/entity-mappers";
+import { fileArtifactsStore } from "../entity-management/file-artifacts-store";
+import { EntityType } from "../entity-management/types";
+
+export async function saveAndRefresh(
+  queryClient: QueryClient,
+  tableName: string,
+  yaml: string
+) {
+  overlay.set({ title: `Importing ${tableName}.yaml` });
```
Comment (ericpgreen2) [inline web-common/src/features/sources/saveAndRefresh.ts:16]: Move this (UI concern) out of the action

**Review by skokenes — APPROVED** [review]
Looks good

**Pull request merged**